Fix Debug settings visibility for vanila flavor-based debug builds

justtwago opened the pull request:
Fixes #19338

The Debug settings screen (on Me tab) should not be visible for apps built with build variants containing vanilla flavor.

To test:
- [ ] Build `jetpackVanillaDebug` app.
- [ ] Log into your account.
- [ ] Go to `Me` tab.
- [ ] Ensure you cannot see the `Debug settings` option on the list.
- [ ] Build any other non-vanilla debug app and make sure the `Debug settings` option is visible on the `Me` tap.
**Upd:**
- [ ] Go to `Login screen` -> `Help (?)` screen and confirm that `Debug settings` is invisible for `jetpackVanillaDebug`-based app
- [ ] Make sure that none of the existing build types have `Debug settings` option on `Me` -> `App settings` screen.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Changes were only made to views.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

Commit:
Debug settings visibility:
Check additional BuildConfig.ENABLE_DEBUG_SETTINGS variable in MeFragment in addition to BuildConfig.DEBUG to handle the debug settings visibility. It should not be visible in vanilla flavor-based debug builds

mkevins reviewed (approved):
Thanks for fixing this @justtwago 👍  I tested this on a Pixel 3a (physical device) on these variants:

- [x] jetpackJalapenoDebug: `Visible`
- [x] jetpackVanillaDebug: `Gone`
- [x] jetpackWasabiDebug: `Visible`
- [x] wordpressJalapenoDebug: `Visible`
- [x] wordpressVanillaDebug: `Gone`
- [x] wordpressWasabiDebug: `Visible`

antonis reviewed (changes requested):
Thank you for wrangling this @justtwago 🙇 
I've tested the change on a Pixel 5/Android 14 and it works as expected for the Me screen.
I noticed two more entry points that we would need to cover:
* Start up screen > Login screen > Help (?): We could add the same condition `&& BuildConfig.ENABLE_DEBUG_SETTINGS` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/accounts/HelpActivity.kt#L116). This entry point was introduced recently with https://github.com/wordpress-mobile/WordPress-Android/pull/18603 to make the settings accessible before login.
* Me > App settings: We [could add a similar fix here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java#L231) or remove this entry point completely as redundant.

Commit:
Remove redundant debug settings option from AppSettingsFragment

Commit:
Debug settings visibility in HelpActivity:
Check additional BuildConfig.ENABLE_DEBUG_SETTINGS variable in HelpActivity in addition to BuildConfig.DEBUG to handle the debug settings visibility. It should not be visible in vanilla flavor-based debug builds.

justtwago commented:
Amazing catch, @antonis! 🚀 
Thank you a lot for pointing out other entry points. I already pushed the suggested changes and updated the testing steps.

antonis reviewed (approved):
Thank you for implementing the changes @justtwago 🙇 
The code LGTM and works as expected 🎉
